File: src/datatypes.ts

export type Height = {
    feet: number;
    inches: number;
    centimeters: number;
}

export type Medication = {
    id: number;
    name: string;
    dosage: string;
    cost_per_pill: number;
}

export type Prescription = {
    id: number;
    medication_id: number;
    medication_name: string;
    quantity: number;
    instructions: string;
    description: string;
    doctor: string;
    frequency: string;
    start_date: string;
    end_date: string;
    refills: number;
    notes: string;
}

export type Patient = {
    id: number;
    name: string;
    address: string;
    phone_number: string;
    date_of_birth: string;
    pregnant: boolean;
    height: Height;
    weight: number;
    allergies: string[];
    prescriptions?: Prescription[]
}

File: src-tauri/src/datatypes.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Height {
    pub feet: u32,
    pub inches: u32,
    pub centimeters: u32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Medication {
    pub id: u64,
    pub name: String,
    pub dosage: String,
    pub frequency: String,
    pub cost_per_pill: f64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Prescription {
    pub id: u64,
    pub medication_id: u64,
    pub medication_name: String,
    pub frequency: String,
    pub start_date: String,
    pub end_date: String,
    pub refills: u32,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Patient {
    pub id: u64,
    pub name: String,
    pub address: String,
    pub phone_number: String,
    pub date_of_birth: String,
    pub pregnant: bool,
    pub height: Height,
    pub weight: f64,
    pub allergies: Vec<String>,
    pub prescriptions: Option<Vec<Prescription>>,
}